packages/stream_chat/lib/src/client/channel.dart

```diff
@@ -1770,6 +1794,19 @@ class ChannelClientState {
     ));
   }
 
+  void _incrementWatcher(User user) {
+    List<User> newWatchers;
+
+    if (channelState.watchers != null) {
+      channelState.watchers!.add(user);
```

Comment:
we should not update the `channelState` properties directly, rather just emit a new state with the update `channelState`.

```dart
updateChannelState(
    channelState.copyWith(
        watchers: updatedWatcher
    )
);
```